lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -195,11 +194,10 @@ public ParquetReader(
         this.writeChecksumBuilder = writeValidation.map(validation -> createWriteChecksumBuilder(validation.getTypes()));
         this.rowGroupStatisticsValidation = writeValidation.map(validation -> createStatisticsValidationBuilder(validation.getTypes()));
 
-        this.blockRowRanges = listWithNulls(this.blocks.size());
-        for (PrimitiveField field : primitiveFields) {
-            ColumnDescriptor columnDescriptor = field.getDescriptor();
-            this.paths.put(ColumnPath.get(columnDescriptor.getPath()), columnDescriptor);
-        }
+        this.blockRowRanges = new FilteredRowRanges[this.blocks.size()];
+        Set<ColumnPath> paths = primitiveFields.stream()
+                .map(field -> ColumnPath.get(field.getDescriptor().getPath()))
+                .collect(toImmutableSet());
```

Comment:
Unrelated change?

Reply:
The `paths` map was introduced as a member variable to this class by the column index feature, it's simplification to a local variable is a related change